// @grammarly/origin/dist/packages/tokens/generated/colors.js
const Blue0 = "#F3F6FF";
const Blue10 = "#D1DBFE";
const Blue20 = "#ADBFF9";
const Blue30 = "#7D99F0";
const Blue40 = "#3E6CF4";
const Blue60 = "#2551DA";
const Blue80 = "#02379E";
const Blue90 = "#000A62";
const Blue100 = "#000A26";
const BlueGray0 = "#F4F4F6";
const BlueGray5 = "#E2E4E9";
const BlueGray10 = "#CDD1DC";
const BlueGray20 = "#C2C6D4";
const BlueGray30 = "#ADB2C3";
const BlueGray35 = "#99A0B3";
const BlueGray40 = "#878DA2";
const BlueGray50 = "#798096";
const BlueGray60 = "#646B81";
const BlueGray70 = "#565B6C";
const BlueGray80 = "#474B58";
const BlueGray90 = "#27282E";
const BlueGray100 = "#161719";
const Gold0 = "#FFF6E0";
const Gold10 = "#FFEBB8";
const Gold20 = "#FFDC85";
const Gold30 = "#FFBF47";
const Gold40 = "#FFA10A";
const Gold60 = "#E57300";
const Gold80 = "#BD5200";
const Gold90 = "#7A3500";
const Gold100 = "#1A0B00";
const Green0 = "#EAFAF9";
const Green10 = "#B1F0E8";
const Green20 = "#73E1D4";
const Green30 = "#2CC9B6";
const Green40 = "#15A694";
const Green60 = "#027E6F";
const Green80 = "#016A5E";
const Green90 = "#014C43";
const Green100 = "#00231F";
const LightGreen = "#00E0AC";
const Magenta0 = "#FFE5F0";
const Magenta10 = "#FEC8DD";
const Magenta20 = "#FAACCB";
const Magenta30 = "#F079A8";
const Magenta40 = "#E14683";
const Magenta60 = "#D21861";
const Magenta80 = "#990C43";
const Magenta90 = "#600428";
const Magenta100 = "#26000F";
const NeutralGray0 = "#F5F5F5";
const NeutralGray5 = "#E6E6E6";
const NeutralGray10 = "#EBEBEB";
const NeutralGray20 = "#D9D9D9";
const NeutralGray30 = "#BCBCBC";
const NeutralGray35 = "#A9A9A9";
const NeutralGray40 = "#A8A8A8";
const NeutralGray50 = "#858585";
const NeutralGray60 = "#707070";
const NeutralGray70 = "#616161";
const NeutralGray80 = "#545454";
const NeutralGray90 = "#2E2E2E";
const NeutralGray100 = "#1C1C1C";
const Purple0 = "#F7F6FE";
const Purple10 = "#DDD9F9";
const Purple20 = "#CAC3F7";
const Purple30 = "#A598F0";
const Purple40 = "#8675EB";
const Purple60 = "#5E47E5";
const Purple80 = "#3D27C0";
const Purple90 = "#1B0D6F";
const Purple100 = "#0E073B";
const Red0 = "#FFEAE9";
const Red10 = "#FFD1D0";
const Red20 = "#FFA8A8";
const Red30 = "#FF7A74";
const Red40 = "#FF4D45";
const Red60 = "#EB0A00";
const Red80 = "#CD0800";
const Red90 = "#8F0600";
const Red100 = "#510300";
const Teal0 = "#E6FDFD";
const Teal10 = "#CCFAFA";
const Teal20 = "#B3F8F8";
const Teal30 = "#80F3F3";
const Teal40 = "#00E6E6";
const Teal60 = "#00C4C4";
const Teal80 = "#027D7D";
const Teal90 = "#005C5C";
const Teal100 = "#001F1F";
const Transparent = "#FFFFFF00";
const White = "#FFFFFF";
const YellowGreen = "#D5FF00";
export {
  Blue0,
  Blue10,
  Blue100,
  Blue20,
  Blue30,
  Blue40,
  Blue60,
  Blue80,
  Blue90,
  BlueGray0,
  BlueGray10,
  BlueGray100,
  BlueGray20,
  BlueGray30,
  BlueGray35,
  BlueGray40,
  BlueGray5,
  BlueGray50,
  BlueGray60,
  BlueGray70,
  BlueGray80,
  BlueGray90,
  Gold0,
  Gold10,
  Gold100,
  Gold20,
  Gold30,
  Gold40,
  Gold60,
  Gold80,
  Gold90,
  Green0,
  Green10,
  Green100,
  Green20,
  Green30,
  Green40,
  Green60,
  Green80,
  Green90,
  LightGreen,
  Magenta0,
  Magenta10,
  Magenta100,
  Magenta20,
  Magenta30,
  Magenta40,
  Magenta60,
  Magenta80,
  Magenta90,
  NeutralGray0,
  NeutralGray10,
  NeutralGray100,
  NeutralGray20,
  NeutralGray30,
  NeutralGray35,
  NeutralGray40,
  NeutralGray5,
  NeutralGray50,
  NeutralGray60,
  NeutralGray70,
  NeutralGray80,
  NeutralGray90,
  Purple0,
  Purple10,
  Purple100,
  Purple20,
  Purple30,
  Purple40,
  Purple60,
  Purple80,
  Purple90,
  Red0,
  Red10,
  Red100,
  Red20,
  Red30,
  Red40,
  Red60,
  Red80,
  Red90,
  Teal0,
  Teal10,
  Teal100,
  Teal20,
  Teal30,
  Teal40,
  Teal60,
  Teal80,
  Teal90,
  Transparent,
  White,
  YellowGreen
};
