// @grammarly/origin/dist/packages/tokens/generated/colors.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const Blue0 = "#F3F6FF";
const Blue10 = "#D1DBFE";
const Blue20 = "#ADBFF9";
const Blue30 = "#7D99F0";
const Blue40 = "#3E6CF4";
const Blue60 = "#2551DA";
const Blue80 = "#02379E";
const Blue90 = "#000A62";
const Blue100 = "#000A26";
const BlueGray0 = "#F4F4F6";
const BlueGray5 = "#E2E4E9";
const BlueGray10 = "#CDD1DC";
const BlueGray20 = "#C2C6D4";
const BlueGray30 = "#ADB2C3";
const BlueGray35 = "#99A0B3";
const BlueGray40 = "#878DA2";
const BlueGray50 = "#798096";
const BlueGray60 = "#646B81";
const BlueGray70 = "#565B6C";
const BlueGray80 = "#474B58";
const BlueGray90 = "#27282E";
const BlueGray100 = "#161719";
const Gold0 = "#FFF6E0";
const Gold10 = "#FFEBB8";
const Gold20 = "#FFDC85";
const Gold30 = "#FFBF47";
const Gold40 = "#FFA10A";
const Gold60 = "#E57300";
const Gold80 = "#BD5200";
const Gold90 = "#7A3500";
const Gold100 = "#1A0B00";
const Green0 = "#EAFAF9";
const Green10 = "#B1F0E8";
const Green20 = "#73E1D4";
const Green30 = "#2CC9B6";
const Green40 = "#15A694";
const Green60 = "#027E6F";
const Green80 = "#016A5E";
const Green90 = "#014C43";
const Green100 = "#00231F";
const LightGreen = "#00E0AC";
const Magenta0 = "#FFE5F0";
const Magenta10 = "#FEC8DD";
const Magenta20 = "#FAACCB";
const Magenta30 = "#F079A8";
const Magenta40 = "#E14683";
const Magenta60 = "#D21861";
const Magenta80 = "#990C43";
const Magenta90 = "#600428";
const Magenta100 = "#26000F";
const NeutralGray0 = "#F5F5F5";
const NeutralGray5 = "#E6E6E6";
const NeutralGray10 = "#EBEBEB";
const NeutralGray20 = "#D9D9D9";
const NeutralGray30 = "#BCBCBC";
const NeutralGray35 = "#A9A9A9";
const NeutralGray40 = "#A8A8A8";
const NeutralGray50 = "#858585";
const NeutralGray60 = "#707070";
const NeutralGray70 = "#616161";
const NeutralGray80 = "#545454";
const NeutralGray90 = "#2E2E2E";
const NeutralGray100 = "#1C1C1C";
const Purple0 = "#F7F6FE";
const Purple10 = "#DDD9F9";
const Purple20 = "#CAC3F7";
const Purple30 = "#A598F0";
const Purple40 = "#8675EB";
const Purple60 = "#5E47E5";
const Purple80 = "#3D27C0";
const Purple90 = "#1B0D6F";
const Purple100 = "#0E073B";
const Red0 = "#FFEAE9";
const Red10 = "#FFD1D0";
const Red20 = "#FFA8A8";
const Red30 = "#FF7A74";
const Red40 = "#FF4D45";
const Red60 = "#EB0A00";
const Red80 = "#CD0800";
const Red90 = "#8F0600";
const Red100 = "#510300";
const Teal0 = "#E6FDFD";
const Teal10 = "#CCFAFA";
const Teal20 = "#B3F8F8";
const Teal30 = "#80F3F3";
const Teal40 = "#00E6E6";
const Teal60 = "#00C4C4";
const Teal80 = "#027D7D";
const Teal90 = "#005C5C";
const Teal100 = "#001F1F";
const Transparent = "#FFFFFF00";
const White = "#FFFFFF";
const YellowGreen = "#D5FF00";
exports.Blue0 = Blue0;
exports.Blue10 = Blue10;
exports.Blue100 = Blue100;
exports.Blue20 = Blue20;
exports.Blue30 = Blue30;
exports.Blue40 = Blue40;
exports.Blue60 = Blue60;
exports.Blue80 = Blue80;
exports.Blue90 = Blue90;
exports.BlueGray0 = BlueGray0;
exports.BlueGray10 = BlueGray10;
exports.BlueGray100 = BlueGray100;
exports.BlueGray20 = BlueGray20;
exports.BlueGray30 = BlueGray30;
exports.BlueGray35 = BlueGray35;
exports.BlueGray40 = BlueGray40;
exports.BlueGray5 = BlueGray5;
exports.BlueGray50 = BlueGray50;
exports.BlueGray60 = BlueGray60;
exports.BlueGray70 = BlueGray70;
exports.BlueGray80 = BlueGray80;
exports.BlueGray90 = BlueGray90;
exports.Gold0 = Gold0;
exports.Gold10 = Gold10;
exports.Gold100 = Gold100;
exports.Gold20 = Gold20;
exports.Gold30 = Gold30;
exports.Gold40 = Gold40;
exports.Gold60 = Gold60;
exports.Gold80 = Gold80;
exports.Gold90 = Gold90;
exports.Green0 = Green0;
exports.Green10 = Green10;
exports.Green100 = Green100;
exports.Green20 = Green20;
exports.Green30 = Green30;
exports.Green40 = Green40;
exports.Green60 = Green60;
exports.Green80 = Green80;
exports.Green90 = Green90;
exports.LightGreen = LightGreen;
exports.Magenta0 = Magenta0;
exports.Magenta10 = Magenta10;
exports.Magenta100 = Magenta100;
exports.Magenta20 = Magenta20;
exports.Magenta30 = Magenta30;
exports.Magenta40 = Magenta40;
exports.Magenta60 = Magenta60;
exports.Magenta80 = Magenta80;
exports.Magenta90 = Magenta90;
exports.NeutralGray0 = NeutralGray0;
exports.NeutralGray10 = NeutralGray10;
exports.NeutralGray100 = NeutralGray100;
exports.NeutralGray20 = NeutralGray20;
exports.NeutralGray30 = NeutralGray30;
exports.NeutralGray35 = NeutralGray35;
exports.NeutralGray40 = NeutralGray40;
exports.NeutralGray5 = NeutralGray5;
exports.NeutralGray50 = NeutralGray50;
exports.NeutralGray60 = NeutralGray60;
exports.NeutralGray70 = NeutralGray70;
exports.NeutralGray80 = NeutralGray80;
exports.NeutralGray90 = NeutralGray90;
exports.Purple0 = Purple0;
exports.Purple10 = Purple10;
exports.Purple100 = Purple100;
exports.Purple20 = Purple20;
exports.Purple30 = Purple30;
exports.Purple40 = Purple40;
exports.Purple60 = Purple60;
exports.Purple80 = Purple80;
exports.Purple90 = Purple90;
exports.Red0 = Red0;
exports.Red10 = Red10;
exports.Red100 = Red100;
exports.Red20 = Red20;
exports.Red30 = Red30;
exports.Red40 = Red40;
exports.Red60 = Red60;
exports.Red80 = Red80;
exports.Red90 = Red90;
exports.Teal0 = Teal0;
exports.Teal10 = Teal10;
exports.Teal100 = Teal100;
exports.Teal20 = Teal20;
exports.Teal30 = Teal30;
exports.Teal40 = Teal40;
exports.Teal60 = Teal60;
exports.Teal80 = Teal80;
exports.Teal90 = Teal90;
exports.Transparent = Transparent;
exports.White = White;
exports.YellowGreen = YellowGreen;
