// @grammarly/origin/dist/packages/tokens/generated/elevation.js
const Elevation100 = {
  boxShadow: "0px 1px 4px 0.5px var(--color-border-base-default-f3)",
  border: "1px solid var(--color-elevation-outline-default-1)"
};
const Elevation200 = {
  boxShadow: "0px 1px 8px 0.5px var(--color-border-base-default-f5)",
  border: "1px solid var(--color-elevation-outline-default-1)"
};
const Elevation300 = {
  boxShadow: "0px 2px 12px 0.5px var(--color-border-base-default-f5)",
  border: "1px solid var(--color-elevation-outline-default-1)"
};
const ElevationLow = {
  boxShadow: "0px 1px 4px 0.5px var(--color-elevation-base-default-f5)",
  border: "1px solid var(--color-elevation-outline-default-1)"
};
const ElevationMedium = {
  boxShadow: "0px 1px 8px 0.5px var(--color-elevation-base-default-f4)",
  border: "1px solid var(--color-elevation-outline-default-1)"
};
const ElevationHigh = {
  boxShadow: "0px 2px 12px 0.5px var(--color-elevation-base-default-f3)",
  border: "1px solid var(--color-elevation-outline-default-1)"
};
export {
  Elevation100,
  Elevation200,
  Elevation300,
  ElevationHigh,
  ElevationLow,
  ElevationMedium
};
