// @grammarly/origin/dist/packages/design-system/src/components/Loader/BrandedLoader.js
import React__default from "react";
import { clsx } from "../../../../../external/clsx@1.2.1/external/clsx/dist/clsx.m.js";
import { useTheme } from "../ThemeProvider/ThemeProvider.js";
/* empty css                  */
import shLoaderPoster from "./sh_loader.png.js";
const BrandedLoader = ({
  size = "medium",
  variant = "default",
  className,
  ...rest
}) => {
  const { theme, mode } = useTheme();
  const isSuperhumanTheme = theme === "superhuman";
  const isDarkMode = mode === "dark";
  const [isMediaReady, setIsMediaReady] = React__default.useState(false);
  const handleMediaReady = React__default.useCallback(() => {
    setIsMediaReady(true);
  }, []);
  return /* @__PURE__ */ React__default.createElement(
    "div",
    {
      className: clsx(
        "gds-branded-loader-container",
        `gds-branded-loader-container-size-${size.toLowerCase()}`,
        `gds-branded-loader-variant-${variant}`,
        className
      ),
      ...rest
    },
    isSuperhumanTheme ? /* @__PURE__ */ React__default.createElement(React__default.Fragment, null, !isMediaReady && /* @__PURE__ */ React__default.createElement(React__default.Fragment, null, /* @__PURE__ */ React__default.createElement("div", { className: "gds-branded-loader-css-background", "aria-hidden": "true" }), /* @__PURE__ */ React__default.createElement(
      "svg",
      {
        className: "gds-branded-loader-shape",
        xmlns: "http://www.w3.org/2000/svg",
        viewBox: "0 0 28 39",
        fill: "none",
        "aria-hidden": "true"
      },
      /* @__PURE__ */ React__default.createElement(
        "path",
        {
          d: "M24.9353 38.6119L14.7345 33.0258C14.1801 32.7196 13.5043 32.7196 12.9499 33.0258L2.74915 38.6119C1.1071 39.5095 -0.682787 37.688 0.262316 36.0829L12.2477 15.7764C12.9605 14.5673 14.7187 14.5673 15.4315 15.7764L27.4169 36.0829C28.362 37.688 26.5774 39.5095 24.93 38.6119H24.9353ZM13.8422 11.7214C17.0894 11.7214 19.7188 9.09728 19.7188 5.86069C19.7188 2.62411 17.0894 0 13.8422 0C10.5951 0 7.9657 2.62411 7.9657 5.86069C7.9657 9.09728 10.5951 11.7214 13.8422 11.7214Z",
          fill: "#421D24"
        }
      )
    )), isDarkMode ? /* @__PURE__ */ React__default.createElement(
      "video",
      {
        className: "gds-branded-loader-video",
        autoPlay: true,
        loop: true,
        muted: true,
        playsInline: true,
        "aria-hidden": "true",
        poster: shLoaderPoster,
        src: "https://s9c0bed5jyahoeew.public.blob.vercel-storage.com/sh_loader_dark-a3ex3Rkp6yUhLwzqLX8YtlLxIieCk4.mp4",
        onCanPlay: handleMediaReady,
        onLoadedData: handleMediaReady
      }
    ) : /* @__PURE__ */ React__default.createElement(
      "video",
      {
        className: "gds-branded-loader-video",
        autoPlay: true,
        loop: true,
        muted: true,
        playsInline: true,
        "aria-hidden": "true",
        poster: shLoaderPoster,
        src: "https://s9c0bed5jyahoeew.public.blob.vercel-storage.com/sh_loader-MZSYQT8mIQBm56HKIH2Mdtw55wgRyk.mp4",
        onCanPlay: handleMediaReady,
        onLoadedData: handleMediaReady
      }
    )) : /* @__PURE__ */ React__default.createElement(
      "svg",
      {
        viewBox: "0 0 128 128",
        fill: `var(--color-icon-base-${variant})`,
        xmlns: "http://www.w3.org/2000/svg"
      },
      /* @__PURE__ */ React__default.createElement(
        "path",
        {
          className: "gds-branded-loader-bobble gds-branded-loader-top-left",
          d: "M59.7334 38.4C59.7334 26.6179 50.1821 17.0667 38.4001 17.0667C26.618 17.0667 17.0667 26.6179 17.0667 38.4C17.0667 50.1821 26.618 59.7333 38.4001 59.7333H59.7334V38.4Z"
        }
      ),
      /* @__PURE__ */ React__default.createElement(
        "path",
        {
          className: "gds-branded-loader-bobble gds-branded-loader-bottom-left",
          d: "M59.7334 98.1333C59.7334 114.628 46.3616 128 29.8667 128C13.3718 128 6.48499e-05 114.628 6.48499e-05 98.1333C6.48499e-05 81.6384 13.3718 68.2667 29.8667 68.2667H59.7334V98.1333Z"
        }
      ),
      /* @__PURE__ */ React__default.createElement(
        "path",
        {
          className: "gds-branded-loader-bobble gds-branded-loader-top-right",
          d: "M68.2666 29.8667C68.2666 13.3718 81.6384 0 98.1333 0C114.628 0 128 13.3718 128 29.8667C128 46.3616 114.628 59.7333 98.1333 59.7333H68.2666V29.8667Z"
        }
      ),
      /* @__PURE__ */ React__default.createElement(
        "path",
        {
          className: "gds-branded-loader-bobble gds-branded-loader-bottom-right",
          d: "M68.2666 89.6C68.2666 101.382 77.8179 110.933 89.5999 110.933C101.382 110.933 110.933 101.382 110.933 89.6C110.933 77.8179 101.382 68.2667 89.5999 68.2667H68.2666V89.6Z"
        }
      )
    )
  );
};
export {
  BrandedLoader
};
