// @grammarly/origin/dist/packages/design-system/src/components/Form/Form.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const React = require("react");
const clsx_m = require("../../../../../external/clsx@1.2.1/external/clsx/dist/clsx.m.cjs");
;/* empty css           */
const _import = require("../../../../../external/@react-aria_utils@3.19.0_react@18.2.0/external/@react-aria/utils/dist/import.cjs");
const Heading = require("../Heading/Heading.cjs");
const _import$1 = require("../../../../../external/@react-aria_visually-hidden@3.8.3_react@18.2.0/external/@react-aria/visually-hidden/dist/import.cjs");
const Text = require("../Text/Text.cjs");
const Flex = require("../Flex/Flex.cjs");
const DEFAULT_MAX_WIDTH = 650;
const FormContext = /* @__PURE__ */ React.createContext("");
const Form = /* @__PURE__ */ React.forwardRef(
  function Form2(props, forwardedRef) {
    const {
      children,
      onSubmit,
      maxWidth = DEFAULT_MAX_WIDTH,
      spacing = "standard",
      className,
      "aria-labelledby": ariaLabelledBy,
      ...rest
    } = props;
    const headingId = _import.useId();
    function handleSubmit(e) {
      e.preventDefault();
      onSubmit(e);
    }
    const ariaLabelledByIds = React.useMemo(() => {
      const providedIds = Array.isArray(ariaLabelledBy) ? ariaLabelledBy.join(" ") : ariaLabelledBy || "";
      return providedIds ? ` ${headingId} ${providedIds}` : headingId;
    }, [ariaLabelledBy, headingId]);
    return /* @__PURE__ */ React.createElement(
      "form",
      {
        ref: forwardedRef,
        className: clsx_m.clsx(`gds-form gds-form-${spacing}`, className),
        style: { maxWidth: `${maxWidth}px` },
        "aria-labelledby": ariaLabelledByIds,
        onSubmit: handleSubmit,
        ...rest
      },
      /* @__PURE__ */ React.createElement(FormContext.Provider, { value: headingId }, children)
    );
  }
);
const FormHeader = (props) => {
  const {
    children,
    heading,
    headingSize = "large",
    headingLevel = "h2",
    description,
    className,
    ...rest
  } = props;
  const headingId = React.useContext(FormContext);
  return /* @__PURE__ */ React.createElement("div", { id: headingId, className: clsx_m.clsx("gds-form-header", className), ...rest }, /* @__PURE__ */ React.createElement(
    Heading.Heading,
    {
      variant: `heading-${headingSize}`,
      as: headingLevel,
      marginBottom: headingSize === "large" ? 1 : void 0
    },
    heading
  ), Boolean(description) && /* @__PURE__ */ React.createElement(Text.Text, { as: "span" }, description), children);
};
const FormRow = ({ children, className, ...rest }) => {
  return /* @__PURE__ */ React.createElement("div", { className: clsx_m.clsx("gds-form-row", className), ...rest }, children);
};
const FormFieldset = /* @__PURE__ */ React.forwardRef(
  function FormFieldset2(props, forwardedRef) {
    const {
      children,
      legend,
      description,
      isRequired,
      isOptional,
      legendIndicatorForRequired = "(Required)",
      legendIndicatorForOptional = "(Optional)",
      ...rest
    } = props;
    const legendContent = /* @__PURE__ */ React.createElement("legend", null, /* @__PURE__ */ React.createElement(Flex.Flex, null, /* @__PURE__ */ React.createElement(Text.Text, { as: "span", variant: "text-medium", weight: "semibold" }, legend), (isRequired === true || isOptional === true) && /* @__PURE__ */ React.createElement(Text.Text, { className: "gds-form-legend-indicator", as: "p", variant: "text-medium" }, /* @__PURE__ */ React.createElement("span", { "aria-hidden": isRequired === true ? "true" : "false" }, isOptional === true && /* @__PURE__ */ React.createElement(_import$1.VisuallyHidden, null, ": "), isRequired === true ? legendIndicatorForRequired : legendIndicatorForOptional))), Boolean(description) && /* @__PURE__ */ React.createElement(Text.Text, { as: "span", variant: "text-xsmall" }, description));
    return /* @__PURE__ */ React.createElement(
      "fieldset",
      {
        ref: forwardedRef,
        className: "gds-form-fieldset",
        "aria-required": isRequired,
        ...rest
      },
      legendContent,
      children
    );
  }
);
const FormFooter = ({ children, ...rest }) => {
  return /* @__PURE__ */ React.createElement("div", { className: "gds-form-footer", ...rest }, children);
};
exports.Form = Form;
exports.FormFieldset = FormFieldset;
exports.FormFooter = FormFooter;
exports.FormHeader = FormHeader;
exports.FormRow = FormRow;
